src/Vault.sol

```diff
@@ -253,17 +254,31 @@ contract Vault {
             _setBaseCurrency(baseCurrency);
         }
         success = getFreeMargin() >= amount;
+        // Update the vault values
+        (address[] memory AllAssetAddresses, uint256[] memory AllAssetIds, uint256[] memory AllAssetAmounts) =
+            generateAssetData();
+        vault.liqThres = IRegistry(registryAddress).getLiquidationThreshold(
+            AllAssetAddresses, AllAssetIds, AllAssetAmounts, vault.baseCurrency
+        );
     }
 
     /**
      * @notice Can be called by authorised applications to close or decrease a margin position.
      * @param baseCurrency The Base-currency in which the margin position is denominated.
      * @dev All values expressed in the base currency of the vault with same number of decimals as the base currency.
      * @return success Boolean indicating if there the margin position is successfully decreased.
+     * @dev Since decreasing margin position is financial activity, liquidation threshold update is done here.
      * @dev ToDo: Function mainly necessary for integration with untrusted protocols, which is not yet implemnted.
      */
-    function decreaseMarginPosition(address baseCurrency, uint256) public view onlyAuthorized returns (bool success) {
+    function decreaseMarginPosition(address baseCurrency, uint256) public onlyAuthorized returns (bool success) {
         success = baseCurrency == vault.baseCurrency;
+
+        // Update the vault values
+        (address[] memory AllAssetAddresses, uint256[] memory AllAssetIds, uint256[] memory AllAssetAmounts) =
```

Comment:
Can we rename this to ` (address[] memory assetAddresses, uint256[] memory assetIds, uint256[] memory assetAmounts)` to be consistent with other generateAssetData's?

Reply:
sure